docs/source/pages/plotting.rst

```diff
@@ -0,0 +1,276 @@
+
+.. code-block:: python
+
+    ax.set_fontsize(fs=20)
+    fig.set_title("This is a nice plot")
+    fig.save_fig("my_awesome_plot.png")
+
+If you want to include a Torchmetrics plot in a bigger figure that have subfigures and subaxises all ``.plot`` methods
+support an optional `ax` argument where you can pass in the subaxises you want the plot to be inserted into:
+
+.. code-block:: python
+
+    # combine plotting of two metrics into one figure
+    fig, ax = plt.subplots(nrows=1, ncols=2)
+    metric1 = Metric1()
+    metric2 = Metric2()
+    for _ in range(num_updates):
+        metric1.update(preds[i], target[i])
+        metric2.update(preds[i], target[i])
+    metric1.plot(ax=ax[0])
+    metric2.plot(ax=ax[1])
+
+**********************
+Plotting a single step
+**********************
+
+At the most basic level the ``.plot`` method can be used to plot the value from a single step. This can be done in two
+ways:
+* Either ``.plot`` method is called with no input, and internally ``metric.compute()`` is called and that value is plotted
+* ``.plot`` is called on a single returned value by the metric, for example from ``metric.forward()``
+
+In both cases it will generate a plot like this (Accuracy as an example):
+
+.. code-block:: python
+
+    metric = torchmetrics.Accuracy(task="binary")
+    for _ in range(num_updates):
+        metric.update(torch.rand(10,), torch.randint(2, (10,)))
+    fig, ax = metric.plot()
+
+.. image:: ../_static/images/binary_accuracy.png
+   :height: 100px
+   :width: 200 px
+   :scale: 50 %
+   :alt: binary accuracy plot
+   :align: right
+
+A single point plot is not that informative in itself, but if available we will try to include additional information
+such as the lower and upper bounds the particular metric can take an if the metric should be minimized or maximized
+to be optimal. This is true for all metrics that return a scalar tensor.
+Some metrics returns multiple values (such as an tensor with multiple elements or an dict of scalar tensors), and in
+that case calling ``.plot`` will return a figure similar to this:
+
+.. code-block:: python
+
+    metric = torchmetrics.Accuracy(task="multiclass", num_classes=3, average=None)
+    for _ in range(num_updates):
+        metric.update(torch.randint(3, (10,)), torch.randint(3, (10,)))
+    fig, ax = metric.plot()
+
+.. image:: ../_static/images/multiclass_accuracy.png
+   :height: 100px
+   :width: 200 px
+   :scale: 50 %
+   :alt: multiclass accuracy plot
+   :align: right
+
+Here, each element is assumed to be an independent metric and plotted as its own point for comparing. The above is true
+for all metrics that returns a scalar tensor, but if the metric returns a tensor with multiple elements then the
+``.plot`` method will return a specialized plot for that particular metric. Take for example the ``ConfusionMatrix``
+metric:
+
+.. code-block:: python
+
+    metric = torchmetrics.ConfusionMatrix(task="multiclass", num_classes=3)
```

Comment:
thinking about using inline plotting?
So a user could also just copy the code and get the same results...